tutorials/dlp-hybrid-inspect/src/main/java/com/example/dlp/HybridInspectSql.java

```diff
@@ -63,104 +64,215 @@
     try {
       cmd = parser.parse(opts, args);
+      if (cmd.getOptionValue(hybridJobNameOption.getOpt(),null)==null &&
+          cmd.getOptionValue(hybridJobTriggerNameOption.getOpt(),null)==null){
+          throw new Exception ("Missing Hybrid Job or Trigger Name");
+      }
     } catch (ParseException e) {
       LOG.log(Level.SEVERE, "Error parsing command options", e);
       formatter.printHelp(HybridInspectSql.class.getName(), opts);
+      System.out.println("You must include either a hybridJobName or hybridJobTriggerName and either parametes for a" +
+                          " data source to scan or an input file to a JSON list of data sources to scan. Please see" +
+                          " the readme for more details");
       System.exit(1);
       return;
     }
 
-    String databasePassword = null;
-    if (cmd.hasOption(secretManagerResourceName.getOpt())) {
-      LOG.info(String.format(">> Retrieving password from Secret Manager (%s)",
-          cmd.getOptionValue(secretManagerResourceName.getOpt())));
-      databasePassword = accessSecretVersion(ServiceOptions.getDefaultProjectId(),
-          cmd.getOptionValue(secretManagerResourceName.getOpt()),
-          cmd.getOptionValue(secretVersion.getOpt(), "latest"));
+    int totalDatabaseCount = 0;
+    int totalTableCount = 0;
+
+    if (cmd.hasOption(sqlOption.getOpt())) {
+      //Read information from command line parameters to scan a single source
+      System.out.println ("Running scan of a single data source.");
+
+      Database tempDB = new Database();
+      tempDB.threadPoolSize = Integer.parseInt(cmd.getOptionValue(threadPoolSizeOption.getOpt(), "1"));
+      tempDB.sampleRowLimit = Integer.parseInt(cmd.getOptionValue(sampleRowLimitOption.getOpt()));
+      tempDB.databaseType = cmd.getOptionValue(sqlOption.getOpt());;
+      tempDB.databaseInstanceDescription = cmd.getOptionValue(databaseInstanceServerDisplayOption.getOpt());
+      tempDB.databaseInstanceServer = cmd.getOptionValue(databaseInstanceServerOption.getOpt());
+      tempDB.databaseName = cmd.getOptionValue(databaseNameOption.getOpt());
+      tempDB.tableName = cmd.getOptionValue(tableNameOption.getOpt());
+      tempDB.databaseUser = cmd.getOptionValue(databaseUserOption.getOpt());
+      tempDB.secretManagerResourceName = cmd.getOptionValue(secretManagerResourceNameOption.getOpt());
+      totalTableCount = inspectSQLDb(tempDB, cmd.getOptionValue(hybridJobNameOption.getOpt(),null), cmd.getOptionValue(hybridJobTriggerNameOption.getOpt(),null), runId);
+      totalDatabaseCount++;
+    }
+    else if (cmd.hasOption(databasesOption.getOpt())) {
+      //Read a list of data sources from JSON input
+      System.out.println (String.format("Running scan of a data sources from external file [%s]",
+          cmd.getOptionValue(databasesOption.getOpt())));
+      LOG.info(String.format("> Retrieving list from (%s)",
+          cmd.getOptionValue(databasesOption.getOpt())));
+      try {
```

Comment:
This try is too broad, it's not clear what exception it's intended to catch. Try to scope this try as narrowly as possible.

Reply:
Fixed. By moving the Gson file load into it's own method this try/catch is no longer needed.  I removed.  Note, I also made the try/catch in the Gson loading method more narrow.